File: codeK(main)/100 Day Challenge/Interviews/containsMostWater.java
class Solution {
    public int maxArea(int[] height) {
        var left = 0;
        var right = height.length-1;
        var area  = 0;

        while(left < right){
            var minHeight = Math.min(height[left], height[right]);
            var width = right - left;
            area = Math.max(area, width * minHeight);

            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }

            
        }
        return area;
    }
}

File: codeKenya S2/Day 5/Lynn Kamau/ContainsMostWater.py
#Contains Most Water Leetcode Challenge
#You are given an integer array height of length n.
#There are n vertical lines drawn such that the two endpoints of the ith line are (i, 0) and (i, height[i]).
#Find two lines that together with the x-axis form a container, such that the container contains the most water.
#Return the maximum amount of water a container can store.
#Notice that you may not slant the container.

def maxArea(height):
    n = len(height)
    max_area = 0

    for i in range(n):
        for j in range(i + 1, n):
            width = j - i
            h = min(height[i], height[j]) 
            area = width * h
            max_area = max(max_area, area)

    return max_area

#Example test
print(maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7])) 